### ISSUE TITLE
Failure running command as non privileged user

### ISSUE THREAD

**Issue by abelboldu** [ansible/ansible#16052]
##### ISSUE TYPE
Bug Report

##### COMPONENT NAME
core

##### ANSIBLE VERSION
```
$ ansible --version
ansible 2.1.0.0
config file =
configured module search path = Default w/o overrides
```

##### SUMMARY
running a command as a non-privileged user causes ansible to fail. It works on a vagrant box with Ansible

##### STEPS TO REPRODUCE
```
- name: Sync Keystone database
  command: keystone-manage db_sync
  become: yes
  become_user: keystone
  retries: "{{ keystone_api_retries }}"
  delay: "{{ keystone_api_retries_delay }}"
```

##### EXPECTED RESULTS
Ansible should run the command despite the user ownership of files created.

##### ACTUAL RESULTS
```
RUNNING HANDLER [openstack-keystone : Sync Keystone database] ******************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "Failed to change ownership of the temporary files Ansible needs to create despite connecting as root.  Unprivileged become user would be unable to read the file."}
    to retry, use: --limit @tests/test.retry
```

**Comment by abelboldu** [comment]
https://travis-ci.org/abelboldu/openstack-keystone/builds/133914351

**Comment by jctanner** [comment]
@abelboldu this is the comment block from the section of code where the error is output ...

```
    def _fixup_perms(self, remote_path, remote_user, execute=False, recursive=True):
        """
        We need the files we upload to be readable (and sometimes executable)
        by the user being sudo'd to but we want to limit other people's access
        (because the files could contain passwords or other private
        information.  We achieve this in one of these ways:

        * If no sudo is performed or the remote_user is sudo'ing to
          themselves, we don't have to change permisions.
        * If the remote_user sudo's to a privileged user (for instance, root),
          we don't have to change permissions
        * If the remote_user is a privileged user and sudo's to an
          unprivileged user then we change the owner of the file to the
          unprivileged user so they can read it.
        * If the remote_user is an unprivieged user and we're sudo'ing to
          a second unprivileged user then we attempt to grant the second
          unprivileged user access via file system acls.
        * If granting file system acls fails we can set the file to be world
          readable so that the second unprivileged user can read the file.
          Since this could allow other users to get access to private
          information we only do this ansible is configured with
          "allow_world_readable_tmpfiles" in the ansible.cfg
        """
```

What is your connection user?

**Comment by abelboldu** [comment]
My connection user was root

**Comment by mplanchard** [comment]
I also am getting this problem when running the following task:

``` yml
- name: install python packages from requirements files
  become: yes
  become_user: "{{ item.user }}"
  pip:
    requirements: "{{ item.requirements }}"
    virtualenv: "{{ item.virtualenv }}"
    virtualenv_python: "{{ item.virtualenv_python }}"
  with_items: "{{ requirements_files }}"
```

I am running on a Vagrantbox with the `bento/ubuntu-16.04`. This does not seem to occur with "real" remote servers, just the vagrant box. I am using the following provisioning block:

``` ruby
  config.vm.provision "ansible_local" do |ansible|
    ansible.provisioning_path = "/ansible"
    ansible.install_mode = :pip
    ansible.verbose = true
    ansible.playbook = "[redacted].yml"
  end
```

Not sure whether this is a Vagrant problem or an ansible problem. I was able to work around it by running the necessary playbook as root and then adjusting permissions.

My ansible version is 2.1.0.0 and my vagrant version is 1.8.3.

**Comment by leifmadsen** [comment]
Same issue, Ansible 2.1.0.0:

```
- name: Ensure XDG_RUNTIME_DIR is set
  blockinfile:
    dest: .bashrc
    create: true
    block: |
      # Make sure XDG_RUNTIME_DIR is set (used by libvirt
      # for creating config and sockets for qemu:///session
      # connections)
      : ${XDG_RUNTIME_DIR:=/run/user/$(id -u)}
      export XDG_RUNTIME_DIR
  become: true
  become_user: "{{ non_root_user }}"
```

```
TASK [provision/remote : Ensure XDG_RUNTIME_DIR is set] ************************
task path: /home/jenkins/tripleo-quickstart/roles/provision/remote/tasks/main.yml:46
Wednesday 22 June 2016  14:18:34 +0000 (0:00:00.874)       0:00:19.528 ******** 
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: jenkins
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp/ansible-tmp-1466605115.32-46199301548957 `" && echo ansible-tmp-1466605115.32-46199301548957="` echo /tmp/ansible-tmp-1466605115.32-46199301548957 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpraj8dl TO /tmp/ansible-tmp-1466605115.32-46199301548957/blockinfile
<127.0.0.1> EXEC /bin/sh -c 'chown -R stack /tmp/ansible-tmp-1466605115.32-46199301548957/ && sleep 0'
fatal: [localhost]: FAILED! => {"failed": true, "msg": "Failed to change ownership of the temporary files Ansible needs to create despite connecting as root.  Unprivileged become user would be unable to read the file."}
```

**Comment by michaelgrosser** [comment]
I'm also having this issue in Ansible 2.1.0.

```
- name: "Initialize new postgresql cluster"
  become: yes
  become_user: "{{ postgresql_admin_user }}"
  command: "/usr/pgsql-{{ postgresql_version }}/bin/initdb -D /var/lib/pgsql/{{ postgresql_version }}/data"
```

Results in:

```
    virtualbox-iso: TASK [postgresql : Initialize new postgresql cluster] **************************
    virtualbox-iso: task path: /tmp/packer-provisioner-ansible-local/roles/postgresql/tasks/install_postgresql.yml:46
    virtualbox-iso: <192.168.56.104> ESTABLISH LOCAL CONNECTION FOR USER: vagrant
    virtualbox-iso: <192.168.56.104> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp/ansible-tmp-1467128031.22-215365346654496 `" && echo ansible-tmp-1467128031.22-215365346654496="` echo /tmp/ansible-tmp-1467128031.22-215365346654496 `" ) && sleep 0'
    virtualbox-iso: <192.168.56.104> PUT /tmp/tmpaCgkkB TO /tmp/ansible-tmp-1467128031.22-215365346654496/command
    virtualbox-iso: <192.168.56.104> EXEC /bin/sh -c 'chown -R postgres /tmp/ansible-tmp-1467128031.22-215365346654496/ && sleep 0'
    virtualbox-iso: fatal: [192.168.56.104]: FAILED! => {"failed": true, "msg": "Failed to change ownership of the temporary files Ansible needs to create despite connecting as root.  Unprivileged become user would be unable to read the file."}
```

This is in an attempt to build a base Vagrant box for users via Packer, but it's reproduceable using Ansible 2.1.0 directly over SSH against the box as well.

**Comment by karljohns0n** [comment]
I had the same issue with Ansible 2.1.0.0. I had to downgrade to 1.9 to make it works again.

```
- hosts: laravelapp
  vars_files:
    - vars.yml
  remote_user: root
  become: yes
  become_user: www
  tasks:
  - name: Pull Market repo
    git: repo={{ repo_gm }} dest={{ dir_staging }} accept_hostkey=yes key_file=/home/www/.ssh/id_rsa
```

`
2016-06-21 13:50:53,066 p=2938 u=root |  fatal: [IP]: FAILED! => {"failed": true, "msg": "Failed to change ownership of the temporary files Ansible needs to create despite connecting as root.  Unprivileged become user would be unable to read the file."}
`

**Comment by jwilm** [comment]
I've encountered this issue setting up a bare metal Ubuntu 14.04 machine (ie not vagrant). Version 2.0.2.0 seems to work OK, though. The issue is present in both 2.1.0.0 and 2.1.1.0.

**Comment by jdoss** [comment]
I had this issue with running commands as the postgres user on Vagrant and I solved it with the following. 

```
- name: detect if vagrant
  stat: path=/etc/sudoers.d/vagrant-nopasswd
  register: vagrant_sudo_file

- name: fix vagrant sudo
  lineinfile: dest=/etc/sudoers.d/vagrant-nopasswd
              state=present
              regexp='^%?vagrant ALL\='
              line='vagrant ALL=(ALL) NOPASSWD:ALL'
              validate='visudo -cf %s'
  when: vagrant_sudo_file.stat.exists == True
```

**Comment by F30** [comment]
I've encountered this error message when trying to become a non-existing user. With an actually existing user (which also needs to have a shell configured to not trigger another error), everything works fine for me.

**Comment by proffalken** [comment]
I had this error because of a typo:

```
---
- hosts: all
  become: true
  become_user: true
  roles:
```
instead of the correct
```
---
- hosts: all
  become: true
  become_user: root
  roles:
```
Sure enough, if you try and connect with a user that doesn't exist, it will fail.

**Comment by hfinucane** [comment]
I've tracked this down, I think. It fails on CentOS 6 systems, with Ansible >= 2.1, without `acl` set as a mount option in /etc/fstab. Changing the mount option fixes the problem, as does setting `allow_world_readable_tmpfiles`.

Further edit- this appears to [be documented](http://docs.ansible.com/ansible/intro_configuration.html#allow-world-readable-tmpfiles)

**Comment by ansibot** [comment]
@abelboldu Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by madhu222** [comment]
@leifmadsen 

How did you solve this issue? I am also getting the issue "Establishing connection with user jenkins"

**Comment by ssbarnea** [comment]
I just encountered this error while trying to run a simple handler and I a bit clueless regarding what could have caused it and how to avoid it.
```
Failed to change ownership of the temporary files Ansible needs to create despite connecting as root. Unprivileged become user would be unable to read the file.
```

Example:
```
  handlers:
    - name: system python changed
      become: yes
      become_user: xxx
      shell: find ~/*/shiningpanda/jobs/ | xargs rm -rf
```

**Comment by ssbarnea** [comment]
I think that we should fix the error message and include a link to the documentation or at least to mention that to perform this operation your need to `allow-world-readable-tmpfiles=true`

https://docs.ansible.com/ansible/latest/intro_configuration.html#allow-world-readable-tmpfiles

This really needs to be fixed as the current error message doesn't clearly indicates why this is happening. 

In fact, I am inclined to believe that Ansible should find a way to avoid getting this failure because is clear that if the user wants to become another user, they really mean it.

**Comment by abadger** [comment]
@ssbarnea need more information on what you're seeing.

To get to the code that is reporting that error it seems like you have to connect as root but root is unable to chown the files in question. C.ALLOW_WORLD_READABLE_TMPFILES shouldn't change that (as the check when you are root and unable to chown comes before the check for WORLD_READABLE_TMPFILES) so I'm trying to puzzle out what is really going on.  If changing the setting of ALLOW_WORLD_READABLE_TMPFILES is the only difference between it working and not working, that sounds like a fortuitous bug (although it might also have security implications so it might be a bug that we have to close).

**Comment by abadger** [comment]
@ssbarnea if you could get in touch with me on IRC, we can discuss this and I can try to reproduce and see what's going on, then I can summarize our findings back in this bug report.

**Comment by abadger** [comment]
@ssbarnea ping?  (I'm around for a few more days but then going on vacation until after New Years).  (My IRC nick is abadger1999)

**Comment by gitqlt** [comment]
ansible 2.5.0
server and client are: Ubuntu 16.04, Python2.7.12

Cannot create (or replace) file, when become to unprivileged user. It seems that `allow_world_readable_tmpfiles = True` is ineffective. Even with `pipelining = True`, the same problem exists with a temporary file: the user cannot read/execute the temporary file
```
chmod u+x /root/.ansible/tmp/...
-rwx------ 1 root root 76768 Apr 19 00:57 /root/.ansible/tmp/...
```

**Comment by gitqlt** [comment]
Problem exists with ansible 2.5.1
`can't open file '/root/.ansible/tmp/ansible-tmp-1524479105.99-234778350704768/stat.py':`
`[Errno 13] Permission denied`

**Comment by gitqlt** [comment]
ansible 2.5.2 also has this
Use of the special become username `admin` results in that ansible drops the `setfacl` method and changes to a root-readonly temporary file with chown-chmod. Then tries to execute it as `sudo admin`, which will fail as my `admin` cannot read the file.
(You can watch that even if there is no admin user on either of the two systems. It changes from setfacl to chown-chmod)

Other become usernames will work with setfacl OK. @abadger ping?

**Comment by gitqlt** [comment]
Even with `ansible_python_interpreter=/usr/bin/python3` I get the same error, when the remote, unprivileged username is `admin`
/usr/bin/python3: can't open file '/root/.ansible/tmp/ansible-tmp-1527964033.24-76591978792337/stat.py': [Errno 13] Permission denied\r\n"

**Comment by bpetit** [comment]
Hi,

I get this error too. With the following task:

```
- name: create postgresql database
  postgresql_db:
    name: "mydb"
  become: true
  become_method: "su"
  become_user: "postgres"
```

ansible version: 2.5.3

target os: debian 9.4

```
TASK [netbox : create postgresql database] **************************************************************************************************************************************************************
fatal: [ec2-myinstance.eu-west-2.compute.amazonaws.com]: FAILED! => {"msg": "Failed to change ownership of the temporary files Ansible needs to create despite connecting as a privileged user. Unprivileged become user would be unable to read the file."}
```
I tried this using both users admin and an ordinary user, with the same result.

**Comment by abadger** [comment]
@gitqlt Your particular problem is different.  Ansible-2.5 added the ability to specify which usernames are considered administrative accounts.  You can configure this via the admin_users setting: http://docs.ansible.com/ansible/devel/plugins/shell/sh.html?highlight=admin_users I'm talking to the other committers about changing the default value of that so admin is no longer in there (perhaps changing it so that only root is in the default value).

Everyone else... I'm unable to reproduce this behaviour.  I'm installing Debian 9.4 right now since that was the last reported problem to see if the exact platform is needed but I somewhat doubt it.  If  someone can figure out what combination of platform and configuration causes this I can use that to work out some sort of fix.

**Comment by bpetit** [comment]
The ami I used on AWS has ID ami-41e80526 on region eu-west-2. It's official Debian 9.4 base image. I'll try to dig into that problem deeper this week.

**Comment by abadger** [comment]
Okay, with a local vm with debian9.4 installed, this playbook worked fine:

```
---
- hosts: debian9-test
  gather_facts: False
  tasks:
  - name: install packages
    apt:
      name:
        - python3-psycopg2
        - postgresql
  - name: create postgresql database
    postgresql_db:
      name: "mydb"
    become: true
    become_method: "su"
    become_user: "postgres"
```
Inventory file:
```

[all:vars]
ansible_python_interpreter=/usr/bin/python3

[test-vms]
debian9-test
```

ansible invocation:
```
[pts/5@peru /srv/ansible/local-playbooks/provisioning]$ ansible-playbook test.yml -k -e 'ansible_user=root' -vv
ansible-playbook 2.5.3 (detached HEAD 778cb5bd8b) last updated 2018/06/05 15:48:05 (GMT -700)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/badger/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /srv/ansible/stable-2.5/lib/ansible
  executable location = /srv/ansible/stable-2.5/bin/ansible-playbook
  python version = 2.7.15 (default, May 15 2018, 15:37:31) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
Using /etc/ansible/ansible.cfg as config file
SSH password: 

PLAYBOOK: test.yml **************************************************************************************************
1 plays in test.yml

PLAY [debian9-test] *************************************************************************************************
META: ran handlers

TASK [install packages] *********************************************************************************************
task path: /srv/ansible/local-playbooks/provisioning/test.yml:5
ok: [debian9-test] => {"cache_update_time": 1528234860, "cache_updated": false, "changed": false}

TASK [create postgresql database] ***********************************************************************************
task path: /srv/ansible/local-playbooks/provisioning/test.yml:10
changed: [debian9-test] => {"changed": true, "db": "mydb"}
META: ran handlers
META: ran handlers

PLAY RECAP **********************************************************************************************************
debian9-test               : ok=2    changed=1    unreachable=0    failed=0 
```

**Comment by basos9** [comment]
I conducted the following tests on Ansibl3 2.5.4, target system Debian 9 (stretch)
trying to execute a command connecting as an unpriviledged sudoer (admin), escalating permission as a another unpriviledged user (user1).
The bottom line is that ansible can not impersonate the second user without the acl utilitiles (setfact and friends).

Also, I think it is a **bug**, when acl is not installed, setting ALLOW_WORLD_READABLE_TMPFILES has no effect, unless you also set admin_users="root".

Below is the detailed report of the tests conducted.

## ENV
```
ansible-playbook 2.5.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/vgouv/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.14 (default, Sep 23 2017, 22:06:14) [GCC 7.2.0]
```

## TEST with acl installed
OK runned 

OUTPUT
```
....
    "stdout_lines": [
        "uid=1003(user1) gid=1003(user1) groups=1003(user1)", 
        "user1", 
        "/home/admin"
    ]

```


## TEST without acl, acl not installed (e.g. setfacl not installed)
FAILED 

**DEBUG** 
```
...
<10.0.11.40> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=10 -o ControlPath=/home/vgouv/.ansible/cp/8ac52224f3 10.0.11.40 '/bin/sh -c '"'"'chmod u+x /var/tmp/ansible-tmp-1529572512.42-181380211250599/ /var/tmp/ansible-tmp-1529572512.42-181380211250599/command.py && sleep 0'"'"''
<10.0.11.40> (0, '', '')
<10.0.11.40> ESTABLISH SSH CONNECTION FOR USER: admin
<10.0.11.40> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=10 -o ControlPath=/home/vgouv/.ansible/cp/8ac52224f3 10.0.11.40 '/bin/sh -c '"'"'chown user1 /var/tmp/ansible-tmp-1529572512.42-181380211250599/ /var/tmp/ansible-tmp-1529572512.42-181380211250599/command.py && sleep 0'"'"''
<10.0.11.40> (1, '', "chown: changing ownership of '/var/tmp/ansible-tmp-1529572512.42-181380211250599/': Operation not permitted\nchown: changing ownership of '/var/tmp/ansible-tmp-1529572512.42-181380211250599/command.py': Operation not permitted\n")
fatal: [10.0.11.40]: FAILED! => {
    "msg": "Failed to change ownership of the temporary files Ansible needs to create despite connecting as a privileged user. Unprivileged become user would be unable to read the file."
}
```

## TEST without acl, with configured ALLOW_WORLD_READABLE_TMPFILES ansbile.cfg
FAILED

**DEBUG**
```
...
<10.0.11.40> ESTABLISH SSH CONNECTION FOR USER: admin
<10.0.11.40> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=10 -o ControlPath=/home/vgouv/.ansible/cp/8ac52224f3 10.0.11.40 '/bin/sh -c '"'"'chmod u+x /var/tmp/ansible-tmp-1529572748.63-51162416374876/ /var/tmp/ansible-tmp-1529572748.63-51162416374876/command.py && sleep 0'"'"''
<10.0.11.40> (0, '', '')
<10.0.11.40> ESTABLISH SSH CONNECTION FOR USER: admin
<10.0.11.40> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=10 -o ControlPath=/home/vgouv/.ansible/cp/8ac52224f3 10.0.11.40 '/bin/sh -c '"'"'chown user1 /var/tmp/ansible-tmp-1529572748.63-51162416374876/ /var/tmp/ansible-tmp-1529572748.63-51162416374876/command.py && sleep 0'"'"''
<10.0.11.40> (1, '', "chown: changing ownership of '/var/tmp/ansible-tmp-1529572748.63-51162416374876/': Operation not permitted\nchown: changing ownership of '/var/tmp/ansible-tmp-1529572748.63-51162416374876/command.py': Operation not permitted\n")
fatal: [10.0.11.40]: FAILED! => {
    "msg": "Failed to change ownership of the temporary files Ansible needs to create despite connecting as a privileged user. Unprivileged become user would be unable to read the file."
}
```


-- ansible.cfg
```
[defaults]
;; WORKAROUND to ALLOW unpriviledged become users to execure 
;; https://github.com/ansible/ansible/issues/16052
allow_world_readable_tmpfiles = True
```


-- TEST config
```
ansible-config dump
....
ALLOW_WORLD_READABLE_TMPFILES(./test-sudo/ansible.cfg) = True
```

## TEST without acl, with configured ALLOW_WORLD_READABLE_TMPFILES with admin_users reduced ansible.cfg
OK

```
10.0.11.40> ESTABLISH SSH CONNECTION FOR USER: admin
<10.0.11.40> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=10 -o ControlPath=/home/vgouv/.ansible/cp/8ac52224f3 10.0.11.40 '/bin/sh -c '"'"'chown user1 /var/tmp/ansible-tmp-1529573681.5-44796767345759/ /var/tmp/ansible-tmp-1529573681.5-44796767345759/command.py && sleep 0'"'"''
<10.0.11.40> (1, '', "chown: changing ownership of '/var/tmp/ansible-tmp-1529573681.5-44796767345759/': Operation not permitted\nchown: changing ownership of '/var/tmp/ansible-tmp-1529573681.5-44796767345759/command.py': Operation not permitted\n")
 [WARNING]: Using world-readable permissions for temporary files Ansible needs to create when becoming an unprivileged user. This may be
insecure. For information on securing this, see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user

<10.0.11.40> ESTABLISH SSH CONNECTION FOR USER: admin
<10.0.11.40> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=10 -o ControlPath=/home/vgouv/.ansible/cp/8ac52224f3 10.0.11.40 '/bin/sh -c '"'"'chmod a+rx /var/tmp/ansible-tmp-1529573681.5-44796767345759/ /var/tmp/ansible-tmp-1529573681.5-44796767345759/command.py && sleep 0'"'"''
<10.0.11.40> (0, '', '')
...
    "stdout_lines": [
        "uid=1003(user1) gid=1003(user1) groups=1003(user1)", 
        "user1", 
        "/home/admin"
    ]

```

ansible.cfg
[defaults]
;; WORKAROUND to ALLOW unpriviledged become users to execure 
;; https://github.com/ansible/ansible/issues/16052
allow_world_readable_tmpfiles = True
;;
admin_users = [u'root', u'toor']



## NOTE 
it seems to be that the affect source code line might be: 
[lib/ansible/plugins/action/__init__.py]( /ansible/ansible/blob/stable-2.5/lib/ansible/plugins/action/__init__.py )
417:
```
       if self._play_context.become and self._play_context.become_user and self._play_context.become_user not in admin_users + [remote_user]:
```
Why does it include remote_user in admin_users, which in turn excludes the ALLOW_WORLD_READABLE_TMPFILES override 


## OTHER INFO
##

## PB
```
- name: Test sudo as non priviledged
  hosts: all
  diff: true # display diff
  vars: 
  tasks: 
  - name: Test command
    shell: id; whoami; pwd
    become: true
    become_user: user1
```




## INV
```
[vagrant_systems]
10.0.11.40

[vagrant_systems:vars]
ansible_user=admin
ansible_become_pass=12345
```



## TARGET SYSTEM INFO :10.0.11.40
```
OS: PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
KERNEL: Linux contrib-stretch 4.9.0-6-amd64 #1 SMP Debian 4.9.82-1+deb9u3 (2018-03-02) x86_64 GNU/Linux
# python --version
Python 2.7.13
```
```
# groups admin
admin : admin sudo
```

**Comment by ymartin59** [comment]
I got something similar and the reason was my target system had not enough memory to start ansible python modules... Hope this helps.

**Comment by ChristianCandelaria** [comment]
I had the same issue, my problem fixes by deleting the **become** option.

```
- name: test
  become: yes --> DELETE
  become_user: "{{ telenet_usr }}"
  command: whoami
  args:
    chdir: /tmp
```

```
- name: test
  become_user: "{{ telenet_usr }}"
  command: whoami
  args:
    chdir: /tmp
```

**Comment by tboerger** [comment]
The issue similar to https://github.com/ansible/ansible/issues/16052#issuecomment-399038452 had been resolved for me with installing the acl package on my Ubuntu target machine.

**Comment by jpeyret** [comment]
confirming same behavior as @basos9 - works with `acl` installed, but fails without it.

### env

ansible 2.9.9
macos
target:  Ubuntu LTS 20.04 w Python 3.8.2

**Comment by tim-seoss** [comment]
FWIW, I saw this error on `devel` today.  The actual error was to try and become a non-existent user, but the erorr message thrown was `Failed to change ownership of the temporary files Ansible needs to create despite connecting as a privileged user.`

Reproduce with this `ansible.cfg`:
```ini
[defaults]
remote_user = root
```
Failing task is:
```yaml
- name: code from git
    become: yes
    become_method: su
    become_user: nonexistantuser
    git:
      repo: 'https://example.com/mycoderepo.git'
      dest: /home/nonexistentuser/code
```

**Comment by s-hertel** [comment]
@tim-seoss That's a separate issue and working as expected (though perhaps we could add some magic to make the error clearer).

It looks like this has been fixed by #41164

**Comment by andrewufrank** [comment]
I encountered a similar error - become_user to an unprivileged user - which did occur, despite that I had set `allow_world_readable_tmpfiles`. The error appeared after the same code had run with an earlier version of ansible (probably 1.9) and appeared with 2.9 -- it seems a  regression error.
The error disappeared after installing ACL on the target system (without any other change), which surprised me.
